bigquery/google/cloud/bigquery/routine.py

```diff
@@ -0,0 +1,502 @@
+            )
+
+    @property
+    def modified(self):
+        """Optional[datetime.datetime]: Datetime at which the routine was
+        last modified (:data:`None` until set from the server).
+
+        Read-only.
+        """
+        value = self._properties.get(self._PROPERTY_TO_API_FIELD["modified"])
+        if value is not None and value != 0:
+            # value will be in milliseconds.
+            return google.cloud._helpers._datetime_from_microseconds(
+                1000.0 * float(value)
+            )
+
+    @property
+    def language(self):
+        """Optional[str]: The language of the routine.
+
+        Defaults to ``SQL``.
+        """
+        return self._properties.get(self._PROPERTY_TO_API_FIELD["language"])
+
+    @language.setter
+    def language(self, value):
+        self._properties[self._PROPERTY_TO_API_FIELD["language"]] = value
+
+    @property
+    def arguments(self):
+        """List[google.cloud.bigquery.routine.RoutineArgument]: Input/output
+        argument of a function or a stored procedure.
+
+        In-place modification is not supported. To set, replace the entire
+        property value with the modified list of
+        :class:`~google.cloud.bigquery.routine.RoutineArgument` objects.
+        """
+        resources = self._properties.get(self._PROPERTY_TO_API_FIELD["arguments"], [])
+        return [RoutineArgument.from_api_repr(resource) for resource in resources]
+
+    @arguments.setter
+    def arguments(self, value):
+        if not value:
+            resource = []
+        else:
+            resource = [argument.to_api_repr() for argument in value]
+        self._properties[self._PROPERTY_TO_API_FIELD["arguments"]] = resource
+
+    @property
+    def return_type(self):
+        """google.cloud.bigquery_v2.types.StandardSqlDataType: Return type of
+        the routine.
+
+        If absent, the return type is inferred from
+        :attr:`~google.cloud.bigquery.routine.Routine.body` at query time in
+        each query that references this routine. If present, then the
+        evaluated result will be cast to the specified returned type at query
+        time.
+
+        See:
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/routines#resource-routine
+        """
+        resource = self._properties.get(self._PROPERTY_TO_API_FIELD["return_type"])
+        if not resource:
+            return resource
+        output = google.cloud.bigquery_v2.types.StandardSqlDataType()
+        output = json_format.ParseDict(resource, output, ignore_unknown_fields=True)
+        return output
+
+    @return_type.setter
+    def return_type(self, value):
+        if value:
+            resource = json_format.MessageToDict(value)
+        else:
+            resource = None
+        self._properties[self._PROPERTY_TO_API_FIELD["return_type"]] = resource
+
+    @property
+    def imported_libraries(self):
+        """List[str]: The path of the imported JavaScript libraries.
```

Comment:
We should be clearer that this is a set of GCS paths e.g. gs://my-bucket/path/to/lib1.js, not js import/require statements.